instrumentation/spring/spring-integration-4.1/javaagent/src/test/java/SpringIntegrationAndRabbitTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.instrumentation.testing.GlobalTraceUtil.runWithSpan;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.opentelemetry.semconv.incubating.MessagingIncubatingAttributes;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class SpringIntegrationAndRabbitTest extends AbstractRabbitProducerConsumerTest {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  SpringIntegrationAndRabbitTest() {
+    super(null);
+  }
+
+  @Test
+  void should_cooperate_with_existing_RabbitMq_instrumentation() {
+    runWithSpan("parent", () -> producerContext.getBean("producer", Runnable.class).run());
+
+    testing.waitAndAssertTraces(
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
+        trace -> {}, // ignore
```

Comment:
im curious as to why we didnt need to do something like this for the groovy versions.

Reply:
there was a more imperative approach

Reply:
i think you could avoid the need for these if you add the `InstrumentationExtention` to `AbstractRabbitProducerConsumerTest` and then wait for and clear those setup spans like:

```java
abstract InstrumentationExtension testing();

private void startRabbit(Class<?> additionalContext) {
    rabbitMqContainer = ...
    rabbitMqContainer.start();

    SpringApplication producerApp = ....
    producerContext = producerApp.run();

    SpringApplication consumerApp = ....
    consumerContext = consumerApp.run();

    testing().waitForTraces(13);
    testing().clearData();
  }
```

and then in each of your tests that extend it, just add:

```java
  @Override
  InstrumentationExtension testing() {
    return testing;
  }
```

Reply:
good idea, just tried that
turned out that the startup wasn't creating those spans